packages/falter-berlin-ssid-changer/files/hotplug.d/30-ssid-changer

```diff
@@ -8,16 +8,35 @@
 . /lib/functions.sh
 . /lib/lib_ssid-changer.sh
 
+# don't run ssid_changer, if the wizard wasn't run yet.
+if [ ! -f /etc/config/ffwizard ]; then
+    log "ffwizard didn't run yet. Cancelling scripts run."
+    exit 0
+fi
+
+# if ssid_changer wasn't configured by wizard, add it to ffwizard-file
+if ! grep ssid_changer /etc/config/ffwizard; then
+    log "No ssid-changer config found. Writing one to /etc/config/ffwizard."
+    uci_add ffwizard ssid_changer ssid_changer
+    uci_set ffwizard ssid_changer enabled 1
+    uci_commit
+fi
+
 # exit early, if ssid-changer is disabled or interface doesn't matter for wifi
 ENABLED=$(uci_get ffwizard ssid_changer enabled)
 if [ $? = 1 ]; then ENABLED=1; fi # default to enbaled, if value not set
-if [ $ENABLED = 0 ]; then
+if [ "$ENABLED" = 0 ]; then
     exit 0
 fi
-if [ "$INTERFACE" != "ffuplink" ] || !(echo "$INTERFACE" | grep -Eq 'tnl_.*'); then
+if [ "$INTERFACE" != "ffuplink" ] && ! (echo "$INTERFACE" | grep -Eq 'tnl_.*'); then
```

Comment:
I don't think this is correct. This will exit if the interface is not ffuplink and not a tunnel. I think you want exactly the opposite, right?

```
if [ "$INTERFACE" = "ffuplink" ] || (echo "$INTERFACE" | grep -Eq 'tnl_.*'); then
```

Reply:
Without this modification it didn't work for me. Hopefully I can explain why it should work this way:

The ffuplink-interface provides the tunnel to the VPN-servers (traffic to the internet). To my knowledge, `tnl_*` interfaces are created by OLSR to form mesh connections between each other node.

So if the hotplug script runs in hotplug mode, we are only interested in changes that happen at these interfaces. Then we check, if there is still internet available. If there happens anything on other interfaces (plugging/unplugging a LAN-Device), we don't care, as it shouldn't affect the uplink.

For the random checks by crond, I override this check by setting INTERFACE an environment-variable, as it would have been done by hotplug.d